File: src/pocketcloud/cloud/network/packet/impl/type/LogType.php
<?php

namespace pocketcloud\cloud\network\packet\impl\type;

use pocketcloud\cloud\util\enum\EnumTrait;

/**
 * @method static LogType INFO()
 * @method static LogType DEBUG()
 * @method static LogType WARN()
 * @method static LogType ERROR()
 * @method static LogType SUCCESS()
 */
final class LogType {
    use EnumTrait;

    protected static function init(): void {
        self::register("info", new LogType("INFO"));
        self::register("debug", new LogType("DEBUG"));
        self::register("warn", new LogType("WARN"));
        self::register("error", new LogType("ERROR"));
        self::register("success", new LogType("SUCCESS"));
    }

    public function __construct(private readonly string $name) {}

    public function getName(): string {
        return $this->name;
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/type/LogType.java
package de.pocketcloud.network.packet.type;

import de.pocketcloud.common.serialization.Writable;

public enum LogType implements Writable<String> {

    INFO,
    WARN,
    ERROR,
    SUCCESS,
    DEBUG;

    @Override
    public String write() {
        return name();
    }
}